libfoundation/src/foundation-number.cpp

```diff
@@ -207,6 +245,56 @@ bool MCNumberParseUnicodeChars(const unichar_t *p_chars, uindex_t p_char_count,
 
 ////////////////////////////////////////////////////////////////////////////////
 
+bool MCNumberFormat(MCNumberRef p_number, MCStringRef& r_string)
+{
+    switch(p_number -> flags & kMCNumberFlagRepMask)
+    {
+        case kMCNumberFlagSignedIntegerRep:
+            return MCStringFormat(r_string, "%d", p_number -> integer);
+        case kMCNumberFlagUnsignedIntegerRep:
+            return MCStringFormat(r_string, "%u", p_number -> unsigned_integer);
+        case kMCNumberFlagDoubleRep:
+            return MCStringFormat(r_string, "%lf", p_number -> real);
+    }
+    
+    MCUnreachable();
+    
+    return false;
+}
+
+bool MCNumberTryToParse(MCStringRef p_string, MCNumberRef& r_number)
+{
+    const char_t *t_chars;
+    t_chars = MCStringGetNativeCharPtr(p_string);
+    
+    char *t_end;
+    integer_t t_integer;
+    t_integer = strtol((const char *)t_chars, &t_end, 10);
```

Comment:
Hmmm - I wonder if we should not consume whitespace at the start of a string? i.e. The make it strict until there's a reason not too argument...